src/checks/ball.js

```diff
@@ -0,0 +1,24 @@
+const _ = require('lodash');
+const ballData = require('../helpers/ball-data.js');
+const pokemonList = _.keys(ballData.legal);
+
+for (const pokemon of pokemonList) {
+  const parsed = [];
+  ballData.ballTypes.forEach((ball, j) => {
+    const data = parseInt(ballData.legal[pokemon].charAt(j), 8);
+    parsed.push({name: ball, legal: !!(data & 4), ha: !!(data & 2), breedable: !!(data & 1)});
+  });
+  ballData.legal[pokemon] = parsed;
+}
+
+module.exports = {
+  description: 'Illegal ball for Pokémon',
+  field: 'ball',
+  test (pkmn) {
+    const pokemon = pkmn.speciesName.toLowerCase();
+    const pokemonData = ballData.legal[pokemon];
```

Comment:
It looks like this only checks whether a ball is legal for a Pokémon. We could probably make the check more powerful by requiring that the ball be legal with a hidden ability if the Pokémon has its hidden ability, or that the ball be legal by breeding if the Pokémon came from an egg. But that can be in a different PR.